### PR TITLE
Catalog interface refactoring

### LINKED ISSUE #73: Refactor Catalog interface
Related in this [discussion](https://github.com/ufcg-lsd/saps-engine/pull/59#discussion_r371792361) by @thiagomanel.

1. delete all unused methods from this interface
2. do we need two addImageTask method? Maybe, we can keep the one that receives SapsImage parameter
3. also, there is no need to use public in the fields declared in an interface.

### PR DESCRIPTION
**Pull request by thiagoyeds**
Fix #73

### CODE REVIEW

**Review by wesleymonte — DISMISSED** [review]

File: src/main/java/org/fogbowcloud/saps/engine/core/database/Catalog.java [hunk 1]
```diff
@@ -4,97 +4,55 @@
...
 import org.fogbowcloud.saps.engine.core.dispatcher.notifier.Ward;
 import org.fogbowcloud.saps.engine.core.model.SapsImage;
 import org.fogbowcloud.saps.engine.core.model.SapsUser;
 import org.fogbowcloud.saps.engine.core.model.enums.ImageTaskState;
 
 public interface Catalog {
 
-	public String NONE = "None";
-	public int UNLIMITED = -1;
+	String NONE = "None";
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/core/database/Catalog.java]: it seems no one is using this NONE variable :)

File: src/main/java/org/fogbowcloud/saps/engine/core/database/Catalog.java [hunk 2]
```diff
@@ -4,97 +4,55 @@
...
+	int UNLIMITED = -1;
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/core/database/Catalog.java]: why all these vars from line 15 to 23 are declared in the interface? aren't they implementation specific?

**Review by thiagomanel — CHANGES_REQUESTED** [review]
I decided to back-off of this review. Please, @wesleymonte give another pass. general comments: this interface still has implementation-specif code (including SQLExceptions being thrown)

File: src/main/java/org/fogbowcloud/saps/engine/core/database/Catalog.java (hunk 1 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/database/Catalog.java]: Variable NONE removed

File: src/main/java/org/fogbowcloud/saps/engine/core/database/Catalog.java (hunk 2 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/database/Catalog.java]: All vars from line 15 to 23 was moved to implementation class

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    List<Ward> getUsersToNotify() throws SQLException;
+
+    void updateImageTask(SapsImage imageTask) throws SQLException;
+
+    boolean isUserNotifiable(String userEmail) throws SQLException;
+
+    List<SapsImage> getAllTasks() throws SQLException;
+
+    List<SapsImage> getTasksInProcessingStates() throws SQLException;
```
Comment (wesleymonte) [inline src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java]: Make this method generic for any state. You can receive a state or a array of states as a parameter, something like `getTasksByState(TaskState... tasksStates)`.

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import java.util.List;
+
+import org.fogbowcloud.saps.engine.core.catalog.exceptions.TaskNotFoundException;
+import org.fogbowcloud.saps.engine.core.catalog.exceptions.UserNotFoundException;
+import org.fogbowcloud.saps.engine.core.dispatcher.notifier.Ward;
+import org.fogbowcloud.saps.engine.core.model.SapsImage;
+import org.fogbowcloud.saps.engine.core.model.SapsUser;
+import org.fogbowcloud.saps.engine.core.model.enums.ImageTaskState;
+
+public interface Catalog {
```
Comment (wesleymonte) [inline src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java:14]: Replace all SQL Exceptions from method signatures by a specific catalog exception, something like `CatalogException`.

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+    int UNLIMITED = -1;
```
Comment (wesleymonte) [inline src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java]: Remove this field from the interface. Create a class to store catalog constants.

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+    SapsImage getTaskById(String taskId) throws SQLException, TaskNotFoundException;
+
+    SapsUser getUserByEmail(String userEmail) throws SQLException, UserNotFoundException;
+
+    void removeNotification(String submissionId, String taskId, String userEmail) throws SQLException;
+
+    void removeStateStamp(String taskId, ImageTaskState state, Timestamp timestamp) throws SQLException;
+
+    List<SapsImage> getSuccessfullyProcessedTasks(String region, Date initDate, Date endDate, String inputGathering,
```
Comment (wesleymonte) [inline src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java]: You make this method generic by passing the task state as a parameter, avoiding create a method for each type of state. I would rename it to `getTasks`, but i don't know if is the best name.

**Review by wesleymonte — CHANGES_REQUESTED** [review]

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java (hunk 4 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java:14]: Done

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java (hunk 5 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java]: Done

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java (hunk 3 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java]: Done

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java (hunk 6 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java]: done

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
+package org.fogbowcloud.saps.engine.core.catalog;
+
+import java.sql.SQLException;
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java]: this import (line #3) is not necessary

Comment (thiagomanel) [reply src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java]: @ThiagoWhispher please, configure your IDE to auto-detect this.

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+import org.fogbowcloud.saps.engine.core.catalog.exceptions.TaskNotFoundException;
+import org.fogbowcloud.saps.engine.core.catalog.exceptions.UserNotFoundException;
+import org.fogbowcloud.saps.engine.core.dispatcher.notifier.Ward;
+import org.fogbowcloud.saps.engine.core.model.SapsImage;
+import org.fogbowcloud.saps.engine.core.model.SapsUser;
+import org.fogbowcloud.saps.engine.core.model.enums.ImageTaskState;
+
+public interface Catalog {
+
+    SapsImage addTask(String taskId, String dataset, String region, Date date, int priority, String user,
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java:59]: this addTask method is complex. please, add a doc to it. I want to one what is the meaning of each variable. Also, tell when does the exception is thrown.

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+                      String inputdownloadingPhaseTag, String digestInputdownloading, String preprocessingPhaseTag,
+                      String digestPreprocessing, String processingPhaseTag, String digestProcessing) throws CatalogException;
+
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java:62]: this method is complex to understand. doc it. can we have multiple changetimestamps per state?

Comment (thiagomanel) [reply src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java:62]: I mean, "add" means we can have multiple

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+    void addStateChangeTime(String taskId, ImageTaskState state, Timestamp timestamp) throws CatalogException;
+
+    void addUser(String userEmail, String userName, String userPass, boolean userState, boolean userNotify,
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java]: addUser is also complex. what is an userState? didn't we decide to remove all Notify stuff? userName and userPass have a max size?

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+                 boolean adminRole) throws CatalogException;
+
+    void addUserNotification(String submissionId, String taskId, String userEmail) throws CatalogException;
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java]: didn't we decide to remove the notification features? If so, why do we still need the addUserNotification and getUsersToNotify methods?

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+
+    List<Ward> getUsersToNotify() throws CatalogException;
+
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java]: this updateImageTask smeels bad. It seems we use it in a single line of our base, `UpdateTaskRetry`, to do the following
```
task.setUpdateTime(imageStore.getTaskById(task.getTaskId()).getUpdateTime());
imageStore.updateImageTask(task);
```

it is ugly because this catalog interface has some method to `SapsImage` object (e.g.  `addStateChangeTime`) but it is also possible to change the state of the catalog by modifying the object and calling the `updateImageTask` method

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+    void updateImageTask(SapsImage imageTask) throws CatalogException;
+
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java:88]: again, didn't we remove notify features?

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+    List<SapsImage> getTasksByState(int limit, ImageTaskState... tasksStates) throws CatalogException;
+
+    SapsImage getTaskById(String taskId) throws CatalogException, TaskNotFoundException;
+
+    SapsUser getUserByEmail(String userEmail) throws CatalogException, UserNotFoundException;
+
+    void removeNotification(String submissionId, String taskId, String userEmail) throws CatalogException;
+
+    void removeStateChangeTime(String taskId, ImageTaskState state, Timestamp timestamp) throws CatalogException;
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java:104]: I think this `removeStateChangeTime`. Why do we need to remove a change time?

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+
+    List<SapsImage> getTasksByFilters(ImageTaskState state, String region, Date initDate, Date endDate, String inputGathering,
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java]: terrible name. did you mean `filterTasks`?

File: src/main/java/org/fogbowcloud/saps/engine/core/archiver/Archiver.java [hunk 16]
```diff
@@ -202,7 +202,7 @@ private void garbageCollector() {
 	 * @throws Exception
 	 */
 	private void cleanUnfinishedArchivedData() throws Exception {
-		List<SapsImage> archivingTasks = tasksInArchivingState(Catalog.UNLIMITED);
+		List<SapsImage> archivingTasks = tasksInArchivingState(CatalogConstants.UNLIMITED);
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/core/archiver/Archiver.java]: what is the point of having the UNLIMITED on the CatalogConstants interfaces? also, it seems odd that we need this limit only in one method of the Catalog interface `getTasksByState`
before modifying the code, please, explain why do we need this. if we really need this in this method, why don't we need in the other ones that return a collection of SapsImage? And, if we really need this, it seems better to have two methods, one that does not have the `limit` parameter and a second one that has (and, please delete this UNLIMITED flag)

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/CatalogConstants.java [hunk 17]
```diff
@@ -0,0 +1,5 @@
+package org.fogbowcloud.saps.engine.core.catalog;
+
+public class CatalogConstants {
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/core/catalog/CatalogConstants.java]: As I described in the comment on the Archiver code, I think we don't need this constant

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/exceptions/CatalogException.java [hunk 18]
```diff
@@ -0,0 +1,10 @@
+package org.fogbowcloud.saps.engine.core.catalog.exceptions;
+
+public class CatalogException extends RuntimeException{
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/core/catalog/exceptions/CatalogException.java]: @ThiagoWhispher, please, configure your IDE to automatically detect problems such `RuntimeException{`

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/exceptions/CatalogException.java [hunk 19]
```diff
@@ -0,0 +1,10 @@
...
+
+    private static final long serialVersionUID = -2520888793776997437L;
+
+    public CatalogException(String msg){
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/core/catalog/exceptions/CatalogException.java]: same problem here. In fact, you've just made a `WesleyProblem`

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/exceptions/TaskNotFoundException.java [hunk 20]
```diff
@@ -0,0 +1,10 @@
+package org.fogbowcloud.saps.engine.core.catalog.exceptions;
+
+public class TaskNotFoundException extends RuntimeException{
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/core/catalog/exceptions/TaskNotFoundException.java]: `WesleyProblem`here

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/exceptions/TaskNotFoundException.java [hunk 21]
```diff
@@ -0,0 +1,10 @@
...
+
+    private static final long serialVersionUID = -2520888793776997437L;
+
+    public TaskNotFoundException(String msg){
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/core/catalog/exceptions/TaskNotFoundException.java]: `WesleyProblem` here too

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/exceptions/UserNotFoundException.java [hunk 22]
```diff
@@ -0,0 +1,10 @@
+package org.fogbowcloud.saps.engine.core.catalog.exceptions;
+
+public class UserNotFoundException extends RuntimeException{
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/core/catalog/exceptions/UserNotFoundException.java]: `WesleyProblem` 2x in this class

File: src/main/java/org/fogbowcloud/saps/engine/utils/retry/catalog/CatalogRetry.java [hunk 23]
```diff
@@ -4,5 +4,5 @@
 
 public interface CatalogRetry<T> {
 
-	public T run() throws SQLException;
+	T run() throws SQLException;
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/utils/retry/catalog/CatalogRetry.java]: it seems odd to throw an SQLException in this class? how is this related to SQL

File: src/main/java/org/fogbowcloud/saps/engine/utils/retry/catalog/GetTaskById.java [hunk 24]
```diff
@@ -16,8 +14,8 @@ public GetTaskById(Catalog imageStore, String taskId) {
 	}
 
 	@Override
-	public SapsImage run() throws SQLException {
-		return imageStore.getTask(taskId);
+	public SapsImage run(){
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/utils/retry/catalog/GetTaskById.java]: `WesleyProblem`

File: src/main/java/org/fogbowcloud/saps/engine/utils/retry/catalog/GetTasksRetry.java [hunk 25]
```diff
@@ -20,8 +19,9 @@ public GetTasksRetry(Catalog imageStore, ImageTaskState state, Integer limit) {
 	}
 	
 	@Override
-	public List<SapsImage> run() throws SQLException {
-		return imageStore.getIn(state, limit);
+	public List<SapsImage> run() {
+		ImageTaskState[] states = {state};
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/utils/retry/catalog/GetTasksRetry.java]: why not declaring and init this variable in the constructor?

File: src/main/java/org/fogbowcloud/saps/engine/utils/retry/catalog/GetUser.java [hunk 26]
```diff
@@ -16,10 +15,10 @@ public GetUser(Catalog imageStore, String userEmail) {
 	}
 
 	@Override
-	public SapsUser run() throws SQLException {
+	public SapsUser run(){
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/utils/retry/catalog/GetUser.java]: `WesleyProblem`

File: src/main/java/org/fogbowcloud/saps/engine/utils/retry/catalog/GetUser.java [hunk 27]
```diff
@@ -16,10 +15,10 @@ public GetUser(Catalog imageStore, String userEmail) {
...
 		try {
-			return imageStore.getUser(userEmail);
-		} catch (SQLException e) {
+			return imageStore.getUserByEmail(userEmail);
+		} catch (CatalogException e) {
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/utils/retry/catalog/GetUser.java]: you should (almost) never do this (catch and do nothing). At least, log it

File: src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/SubmissionDispatcher.java [hunk 28]
```diff
@@ -31,7 +33,7 @@ public SubmissionDispatcher(Catalog imageStore) {
 	}
 
 	public SubmissionDispatcher(Properties properties) throws SQLException {
-		this.imageStore = new JDBCImageDataStore(properties);
+		this.imageStore = new JDBCCatalog(properties);
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/SubmissionDispatcher.java]: why are we still using `imageStore`? please, rename to `this.catalog`

**Review by thiagomanel — CHANGES_REQUESTED** [review]
thanks, @ThiagoWhispher and @wesleymonte. I think all the request are made are doable. Anyway, be free to poke me away from the keyboard.

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java (hunk 7 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java]: Done

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java (hunk 15 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java]: Yes, lgtm. Done

File: src/main/java/org/fogbowcloud/saps/engine/utils/retry/catalog/GetUser.java (hunk 26 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/utils/retry/catalog/GetUser.java]: Done

File: src/main/java/org/fogbowcloud/saps/engine/utils/retry/catalog/GetTaskById.java (hunk 24 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/utils/retry/catalog/GetTaskById.java]: Done

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/exceptions/UserNotFoundException.java (hunk 22 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/catalog/exceptions/UserNotFoundException.java]: Done

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/exceptions/TaskNotFoundException.java (hunk 21 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/catalog/exceptions/TaskNotFoundException.java]: Done

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/exceptions/TaskNotFoundException.java (hunk 20 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/catalog/exceptions/TaskNotFoundException.java]: Done

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/exceptions/CatalogException.java (hunk 19 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/catalog/exceptions/CatalogException.java]: Done

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/exceptions/CatalogException.java (hunk 18 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/catalog/exceptions/CatalogException.java]: Done

File: src/main/java/org/fogbowcloud/saps/engine/utils/retry/catalog/GetUser.java (hunk 27 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/utils/retry/catalog/GetUser.java]: Done

File: src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/SubmissionDispatcher.java (hunk 28 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/SubmissionDispatcher.java]: Done

File: src/main/java/org/fogbowcloud/saps/engine/utils/retry/catalog/CatalogRetry.java (hunk 23 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/utils/retry/catalog/CatalogRetry.java]: I created an exception class (CatalogRetryException) to represent errors coming from that interface

File: src/main/java/org/fogbowcloud/saps/engine/utils/retry/catalog/GetTasksRetry.java (hunk 25 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/utils/retry/catalog/GetTasksRetry.java]: I moved this declaration to the constructor and only use the state array in the run method

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/CatalogConstants.java (hunk 17 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/catalog/CatalogConstants.java]: I removed the constant related to the catalog and corrected the code so as not to use a limiter because in all cases it was not useful, believing it was an early optimization.

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java (hunk 13 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java:88]: The notification codes are linked with some methods of the Dispatcher component that would be solved in problem #74, but I went ahead and removed it soon.

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java (hunk 11 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java]: Yes, it was decided. I removed all notification-related methods.

File: src/main/java/org/fogbowcloud/saps/engine/core/archiver/Archiver.java (hunk 16 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/archiver/Archiver.java]: I analyzed the code and this constant was only useful in the old SAPS design by the InputDownloader component (which currently no longer exists) in the step of selecting SAPS tasks due to the disk space used by the downloaded or processed data, making it select up to X depending the situation.

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java (hunk 9 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java:62]: At the moment, for each state there is only one changetimestamp, but this is done through removals and additions instead of updates.

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java (hunk 8 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java:59]: Doc added

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java (hunk 10 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java]: Doc added.
`userState` is related to whether the user can authenticate to the SAPS platform, when a new user is created through the dashboard, it is set to false, requiring an admin to enable it.
`userNotify` is related to whether the user wants to receive notifications about their SAPS tasks, but at the moment there is no feature of that.
There is no maximum size for `userName` or `userPass` parameters

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java (hunk 12 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java]: This is a bit confusing, we should review a good design for this situation.

File: src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java (hunk 14 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java:104]: This is somewhat related to this [comment](https://github.com/ufcg-lsd/saps-engine/pull/76#discussion_r375430904)

File: src/main/java/org/fogbowcloud/saps/engine/core/archiver/Archiver.java [hunk 29]
```diff
@@ -22,547 +22,540 @@
...
-		return false;
-	}
-
-	private String getContainerName() {
-		return properties.getProperty(SapsPropertiesConstants.SWIFT_CONTAINER_NAME);
-	}
+    private final Properties properties;
+    private final Catalog catalog;
+    private final SwiftAPIClient swiftAPIClient;
+    private ScheduledExecutorService sapsExecutor;
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/core/archiver/Archiver.java]: shouldn't this `ScheduledExecutorService` var be declared final?

File: src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/SubmissionDispatcher.java [hunk 30]
```diff
@@ -19,480 +18,438 @@
...
+
+    public SubmissionDispatcher(Catalog catalog) {
+        this.catalog = catalog;
+    }
+
+    public SubmissionDispatcher(Properties properties) throws SQLException {
+        this.catalog = new JDBCCatalog(properties);
+    }
+
+    /**
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/SubmissionDispatcher.java]: a reminder: there is no need to doc methods like this `getCatalog`. It is quite obvious. (since we doc it, fine. keep the doc)

File: src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/SubmissionDispatcher.java [hunk 31]
```diff
@@ -19,480 +18,438 @@
...
-
-		for (String region : regions) {
-			List<SapsImage> tasksInCurrentRegion = getProcessedTasksInCatalog(region, initDate, endDate,
-					inputdownloadingPhaseTag, preprocessingPhaseTag, processingPhaseTag);
-			filteredTasks.addAll(tasksInCurrentRegion);
-		}
-		return filteredTasks;
-	}
+
+    private Catalog catalog;
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/SubmissionDispatcher.java]: final?

File: src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/restlet/resource/BaseResource.java [hunk 32]
```diff
@@ -33,7 +33,7 @@ protected boolean authenticateUser(String userEmail, String userPass, boolean mu
 
 		SapsUser user = application.getUser(userEmail);
 		String md5Pass = DigestUtils.md5Hex(userPass);
-		if (user != null && user.getUserPassword().equals(md5Pass) && user.getActive()) {
+		if (user != null && user.getUserPassword().equals(md5Pass) && user.getEnable()) {
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/restlet/resource/BaseResource.java]: this name is not ok. I think you `isEnabled`?

File: src/main/java/org/fogbowcloud/saps/engine/core/model/SapsUser.java [hunk 33]
```diff
@@ -34,8 +31,8 @@ public String getUserPassword() {
 		return userPassword;
 	}
 	
-	public boolean getActive() {
-		return active;
+	public boolean getEnable() {
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/core/model/SapsUser.java]: bad naming

**Review by thiagomanel — CHANGES_REQUESTED** [review]
thank @ThiagoWhispher 
I've added simples requests now.  I think we are able to close this PR after your next commit.

File: src/main/java/org/fogbowcloud/saps/engine/core/archiver/Archiver.java (hunk 29 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/archiver/Archiver.java]: Done

File: src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/SubmissionDispatcher.java (hunk 30 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/SubmissionDispatcher.java]: Remove it.

File: src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/SubmissionDispatcher.java (hunk 31 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/SubmissionDispatcher.java]: Yes, I added.

File: src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/restlet/resource/BaseResource.java (hunk 32 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/restlet/resource/BaseResource.java]: `isEnabled` gets better, I changed it.

File: src/main/java/org/fogbowcloud/saps/engine/core/model/SapsUser.java (hunk 33 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/model/SapsUser.java]: Done

**Comment by thiagoyeds** [comment]
@thiagomanel it's ok for you?

**Review by thiagomanel — APPROVED** [review]
thanks @ThiagoWhispher.  I thinks it is ok, for now. Further refactors should be drive by the FIXME comments.
